test/spmd/test_xla_spmd_python_api_interaction.py

```diff
@@ -123,8 +123,20 @@ def test_runtime_spmd_api(self):
     self.assertTrue(xr.is_spmd())
     del os.environ["XLA_USE_SPMD"]
     self.assertFalse(xr.is_spmd())
-    # reset for other test cases
-    os.environ["XLA_USE_SPMD"] = "1"
+
+    # unittest process can persist XLA_USE_SPMD from other test suites,
+    # so t may be on a SPMD or non-SPMD device. If this test is run independently
+    # outside unittest, then it lives on a non-SPMD device.
+    t = torch.ones(2, 2).to(xm.xla_device())
+
+    # Should enable SPMD without crashing.
+    xr.use_spmd()
+    self.assertTrue("SPMD" in torch_xla._XLAC._xla_get_device_hw_type(t))
+    self.assertTrue(xr.is_spmd())
+
+    # execute replicated
+    self.assertTrue(
+        "{replicated}" in torch_xla._XLAC._get_xla_tensors_hlo([t + t]))
```

Comment:
Could we also add a test which initializes tensors on non-SPMD, calls `use_spmd`, and runs an actual computation on the result?

Reply:
that test can be done after we make device mutable.

Reply:
I think this can be more properly done once we change the device mutable.